### PR TITLE
feat: add definition for `func-deploy` Tekton Task

### PR DESCRIPTION
**Pull request by zroubalik**
Signed-off-by: Zbynek Roubalik <zroubali@redhat.com>

# Changes

-  :gift: add definition for `func-deploy` Tekton Task

/kind enhancement

Relates: #620

### CODE REVIEW

File: pipelines/resources/tekton/task/func-deploy/0.1/func-deploy.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/categories: CLI
+    tekton.dev/tags: cli
+    tekton.dev/platforms: "linux/amd64"
+spec:
+  description: >-
+    This Task performs deploy operation using Knative `func` CLI.
```
Comment (lance) [inline pipelines/resources/tekton/task/func-deploy/0.1/func-deploy.yaml:14]: minor nit
```suggestion
    This Task performs a deploy operation using the Knative `func` CLI.
```

File: pipelines/resources/tekton/task/func-deploy/0.1/func-deploy.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  params:
+  - name: path
+    description: Path where the function project lives.
```
Comment (lance) [inline pipelines/resources/tekton/task/func-deploy/0.1/func-deploy.yaml:17]:
```suggestion
    description: Path to the function project
```
suggestion

File: pipelines/resources/tekton/task/func-deploy/0.1/func-deploy.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    default: ""
+  workspaces:
+    - name: source
+      description: The workspace containing the `func` project.
```
Comment (lance) [inline pipelines/resources/tekton/task/func-deploy/0.1/func-deploy.yaml:21]: Should probably be consistent with above (or change above to use `func`).
```suggestion
      description: The workspace containing the function project
```

**Review by lance — COMMENTED** [review]
Some minor wording suggestions, but otherwise lgtm

**Comment by zroubalik** [comment]
> Some minor wording suggestions, but otherwise lgtm

Feedback incorporated. That's definitely better, thanks!

**Pull request merged**